src/util/psbt/map/global.rs

```diff
@@ -188,125 +188,124 @@ impl Map for PartiallySignedTransaction {
     }
 }
 
-impl PartiallySignedTransaction {
-    pub(crate) fn consensus_decode_global<D: io::Read>(d: D) -> Result<Self, encode::Error> {
-        let mut d = d.take(MAX_VEC_SIZE as u64);
-        let mut tx: Option<Transaction> = None;
-        let mut version: Option<u32> = None;
-        let mut unknowns: BTreeMap<raw::Key, Vec<u8>> = Default::default();
-        let mut xpub_map: BTreeMap<ExtendedPubKey, (Fingerprint, DerivationPath)> = Default::default();
-        let mut proprietary: BTreeMap<raw::ProprietaryKey, Vec<u8>> = Default::default();
-
-        loop {
-            match raw::Pair::consensus_decode(&mut d) {
-                Ok(pair) => {
-                    match pair.key.type_value {
-                        PSBT_GLOBAL_UNSIGNED_TX => {
-                            // key has to be empty
-                            if pair.key.key.is_empty() {
-                                // there can only be one unsigned transaction
-                                if tx.is_none() {
-                                    let vlen: usize = pair.value.len();
-                                    let mut decoder = Cursor::new(pair.value);
-
-                                    // Manually deserialized to ensure 0-input
-                                    // txs without witnesses are deserialized
-                                    // properly.
-                                    tx = Some(Transaction {
-                                        version: Decodable::consensus_decode(&mut decoder)?,
-                                        input: Decodable::consensus_decode(&mut decoder)?,
-                                        output: Decodable::consensus_decode(&mut decoder)?,
-                                        lock_time: Decodable::consensus_decode(&mut decoder)?,
-                                    });
-
-                                    if decoder.position() != vlen as u64 {
-                                        return Err(encode::Error::ParseFailed("data not consumed entirely when explicitly deserializing"))
-                                    }
-                                } else {
-                                    return Err(Error::DuplicateKey(pair.key).into())
+pub fn consensus_decode_global<D: io::Read>(d: D) -> Result<PartiallySignedTransaction, encode::Error> {
```

Comment:
Why remove the pub(crate) visibilty? This is a now public function inside a private mod.
Would it be better to have a pub(crate) or pub(in `Path`)? I don't know if pub(in path) is supported in MSRV.

Reply:
Fair point, I tried with `pub(in path)`. Seems my new shell function to build with 1.29 doesn't work today so I'm relying on CI to see if it works.

Reply:
The first time in my rust experience I finally see the case for this strange `in path` visibility. Usually you need a cross-mod visibility and `in path` is not applicable there...